geocoding.go

```diff
@@ -40,6 +46,38 @@ type ForwardGeocodeRequest struct {
 	Types        Types
 }
 
+func (r ForwardGeocodeRequest) MarshalJSON() ([]byte, error) {
+	type forwardGeocodeRequest struct {
+		SearchText   string       `json:"q,omitempty"`
```

Comment:
Check for unintended single-letter variables, as they may reduce code readability. `q` can also be interpreted in some cases for query. Would you like to  be `searchtext` or `q`? Please check again and confirm.